documentation/docs/05-misc/02-typescript.md

```diff
@@ -2,6 +2,196 @@
 title: TypeScript
 ---
 
+You can use TypeScript to type your Svelte components. The Svelte VSCode extension and the [`svelte-check` CLI](https://www.npmjs.com/package/svelte-check) will check your code for errors and provide hints about how to fix them.
+
+## Setup
+
+Install `typescript` and `svelte-preprocess`:
```

Comment:
we should probably add a heading here and then swap the order of the vite and non-vite sections (I can't easily propose that in the UI) so that the Vite section comes first as most users will be using Vite

```suggestion

### Without Vite

Install `typescript` and `svelte-preprocess`:
```